mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -218,22 +234,14 @@ private[spark] object BLAS extends Serializable {
   def scal(a: Double, x: Vector): Unit = {
     x match {
       case sx: SparseVector =>
-        f2jBLAS.dscal(sx.values.length, a, sx.values, 1)
+        getBLAS(sx.size).dscal(sx.values.length, a, sx.values, 1)
```

Comment:
Not .size, but .values.length here. The actual array is smaller than the size, usually.

Reply:
Resolved in commit https://github.com/apache/spark/pull/27546/commits/118a0d8364641197ccf6e141ae5942309cd27909.